test/functional/helper.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright (c) 2015 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package functional
+
+import (
+	"testing"
+
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/test"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vim25/types"
+	"golang.org/x/net/context"
+)
+
+type Helper struct {
+	*testing.T
+
+	c   *vim25.Client
+	f   *find.Finder
+	fns []func()
+}
+
+func NewHelper(t *testing.T) *Helper {
+	h := &Helper{
+		T: t,
+
+		c:   test.NewAuthenticatedClient(t),
+		fns: make([]func(), 0),
+	}
+
+	h.f = find.NewFinder(h.c, true)
+
+	return h
+}
+
+func (h *Helper) Defer(fn func()) {
```

Comment:
I added it so that resources that are added on-demand from the helper code can be cleaned up as well. Without a teardown function it is not possible to capture those in the scope of the test function. I didn't end up using this in the single test case but think it would be good to leave them in for future stuff. For example: if we want to create a resource pool on demand for every test, we should just do that from the helper, but still destroy it after every test run.